enhancements/network/netflow.md

```diff
@@ -0,0 +1,121 @@
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Collecting all traffic from a cluster at a packet level would be an enormous amount of data.
+An alternative approach for analysis of network traffic is to capture only metadata about the traffic in a cluster.
+This metadata would include information like the protocols, source and destination addresses, port numbers, and the amount of traffic.
+Collecting this metadata becomes a more manageable approach to getting visibility into network activity across the cluster.
+
+There are existing protocols for how this metadata is transmitted between systems, including NetFlow, IPFIX, and sFlow.
+Open vSwitch (OVS) supports these protocols and OVS is a fundamental building block of the OVN-Kubernetes network type.
+
+This proposal discusses how we start providing this functionality by allowing cluster administrators to enable the export of network flow metadata using one of these protocols.
+
+## Motivation
+
+* Cluster administrators need to be able to get network flows out of OpenShift to be consumed by their collectors.
+
+### Goals
+
+- Add support for exporting the network flows traffic in an OpenShift cluster by leveraging the NetFlow/sflow/IPFIX protocols supported by OVS.
+
+### Non-Goals
+
+- This document only applies to OVN-Kubernetes, not OpenShift SDN or any other other network plugin.
+- This document will not discuss the flow logs store solutions. Typically, customers store the NetFlow data in a data warehouse system (e.g. ClickHouse) or search and analytics system (e.g. ElasticSearch). It is out of the scope of this document to discuss which option should be used and if we should manage that solution.
+- This document will not discuss the aggregation of flow data with corresponding K8S pods.
+- This document will not discuss flow visualization or flow dashboards.
+
+## Proposal
+
+OVS supports NetFlow v5, IPFIX and sflow.
+NetFlow v5 does not support IPv6, thus it should not be considered as a default protocol
+sFlow is the most performant protocol as it is a sampling protocol, which polls periodically interfaces to get samples of packets. The drawback is that it does not capture all packets.
+IPFIX is capable of capturing all traffic at line rate and supports IPv6, therefore it should be the default protocol if this option is not specified.
+
+The Cluster Network Operator (CNO) would expose:
+* *exportNetworkFlowsConfig* dict to contain the options to export the network flows
+  * *exportNetworkFlowsProtocol* to specify which protocol should be used. Valid values should be *netflow*, *ipfix* and *sflow*, being the default ipfix.
+  * *exportNetworkFlowsCollectors* list to contain the collectors IPs and ports
```

Comment:
maybe add a note clarifying that specifying more than one IP here means the flows are sent to ALL of them

Reply:
k, will do

Reply:
done